<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsFieldCalculatorBase</class>
 <widget class="QDialog" name="QgsFieldCalculatorBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>681</width>
    <height>624</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Field Calculator</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0" colspan="2">
    <layout class="QVBoxLayout" name="vLayout"/>
   </item>
   <item row="1" column="0">
    <widget class="QCheckBox" name="mOnlyUpdateSelectedCheckBox">
     <property name="text">
      <string>Only update selected features</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox" name="mNewFieldGroupBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Create a new field</string>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <layout class="QGridLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SizeConstraint::SetMinimumSize</enum>
      </property>
      <property name="leftMargin">
       <number>3</number>
      </property>
      <property name="topMargin">
       <number>9</number>
      </property>
      <property name="rightMargin">
       <number>3</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <property name="verticalSpacing">
       <number>3</number>
      </property>
      <item row="5" column="3">
       <widget class="QgsSpinBox" name="mOutputFieldPrecisionSpinBox">
        <property name="toolTip">
         <string>Maximum number of digits after the decimal place. For example 123.45 requires a field precision of 2.</string>
        </property>
        <property name="value">
         <number>2</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="mFieldNameLabel">
        <property name="text">
         <string>Output field name</string>
        </property>
        <property name="buddy">
         <cstring>mOutputFieldNameLineEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="mOutputFieldWidthLabel">
        <property name="toolTip">
         <string>Total length of field (including the number of digits after the decimal place for decimal fields).&lt;br&gt;For example 123.45 requires a decimal field length of 5, and 123456 requires an integer field length of 6.</string>
        </property>
        <property name="text">
         <string>Output field length</string>
        </property>
        <property name="buddy">
         <cstring>mOutputFieldWidthSpinBox</cstring>
        </property>
       </widget>
      </item>
      <item row="4" column="1" colspan="3">
       <widget class="QComboBox" name="mOutputFieldTypeComboBox"/>
      </item>
      <item row="5" column="1">
       <widget class="QgsSpinBox" name="mOutputFieldWidthSpinBox">
        <property name="toolTip">
         <string>Total length of field (including the number of digits after the decimal place for decimal fields).&lt;br&gt;For example 123.45 requires a decimal field length of 5, and 123456 requires an integer field length of 6.</string>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="value">
         <number>15</number>
        </property>
       </widget>
      </item>
      <item row="3" column="1" colspan="3">
       <widget class="QLineEdit" name="mOutputFieldNameLineEdit"/>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="mOutputFieldTypeLabel">
        <property name="text">
         <string>Output field type</string>
        </property>
        <property name="buddy">
         <cstring>mOutputFieldTypeComboBox</cstring>
        </property>
       </widget>
      </item>
      <item row="5" column="2">
       <widget class="QLabel" name="mOutputFieldPrecisionLabel">
        <property name="toolTip">
         <string>Maximum number of digits after the decimal place. For example 123.45 requires a field precision of 2.</string>
        </property>
        <property name="text">
         <string>Precision</string>
        </property>
        <property name="buddy">
         <cstring>mOutputFieldPrecisionSpinBox</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="4">
       <widget class="QCheckBox" name="mCreateVirtualFieldCheckbox">
        <property name="toolTip">
         <string>&lt;p&gt;A virtual field will be recalculated every time it is used. Its definition will be saved in the project file. It will not be saved in the dataprovider and therefore its values not be available in other software.&lt;/p&gt;</string>
        </property>
        <property name="text">
         <string>Create virtual field</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QGroupBox" name="mUpdateExistingGroupBox">
     <property name="title">
      <string>Update existing field</string>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="leftMargin">
       <number>3</number>
      </property>
      <property name="topMargin">
       <number>9</number>
      </property>
      <property name="rightMargin">
       <number>3</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QComboBox" name="mExistingFieldComboBox"/>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="QgsExpressionBuilderWidget" name="builder" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QWidget" name="widget" native="true">
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="mInfoIcon">
        <property name="minimumSize">
         <size>
          <width>64</width>
          <height>64</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>64</width>
          <height>64</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="mEditModeAutoTurnOnLabel">
        <property name="text">
         <string>You are editing information on this layer but the layer is currently not in edit mode. If you click OK, edit mode will automatically be turned on.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="mOnlyVirtualFieldsInfoLabel">
        <property name="text">
         <string>This layer does not support adding new provider fields. You can only add virtual fields.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="5" column="0" colspan="2">
    <widget class="QDialogButtonBox" name="mButtonBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>3</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::StandardButton::Apply|QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Help|QDialogButtonBox::StandardButton::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsSpinBox</class>
   <extends>QSpinBox</extends>
   <header>qgsspinbox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsExpressionBuilderWidget</class>
   <extends>QWidget</extends>
   <header>qgsexpressionbuilderwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>mOnlyUpdateSelectedCheckBox</tabstop>
  <tabstop>mNewFieldGroupBox</tabstop>
  <tabstop>mCreateVirtualFieldCheckbox</tabstop>
  <tabstop>mOutputFieldNameLineEdit</tabstop>
  <tabstop>mOutputFieldTypeComboBox</tabstop>
  <tabstop>mOutputFieldWidthSpinBox</tabstop>
  <tabstop>mOutputFieldPrecisionSpinBox</tabstop>
  <tabstop>mUpdateExistingGroupBox</tabstop>
  <tabstop>mExistingFieldComboBox</tabstop>
  <tabstop>mButtonBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>mButtonBox</sender>
   <signal>accepted()</signal>
   <receiver>QgsFieldCalculatorBase</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>679</x>
     <y>559</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mButtonBox</sender>
   <signal>rejected()</signal>
   <receiver>QgsFieldCalculatorBase</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>679</x>
     <y>559</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
